enclave-runner/src/usercalls/mod.rs

```diff
@@ -371,13 +628,11 @@ impl EnclaveState {
             Err(EnclaveAbort::Secondary) |
             Ok(_) => false,
         };
-
         let cmd = enclave.kind.as_command().unwrap();
         let mut cmddata = cmd.data.lock().unwrap();
-
-        cmddata.threads.clear();
+        //clear the threads_queue
+        cmddata.threads_queue  = crossbeam::queue::SegQueue::new();
         enclave.abort_all_threads();
-
         while cmddata.running_secondary_threads > 0 {
             cmddata = cmd.wait_secondary_threads.wait(cmddata).unwrap();
```

Comment:
This wait logic probably needs to be revised because of the event loop. I'm kind of expecting that waiting on other threads is supposed to happen in the event loop and that no more waiting needs to happen after the event loop exits.